### DISCUSSION TITLE
Which RCX extractor rules to use?

### DISCUSSION THREAD

**Discussion by vijayank88** [The-OpenROAD-Project/OpenLane#1477] [Q&A] [answered]
In `sky130A` PDK selection RCX extraction rules follows `caliber`
```
set ::env(RCX_RULES) {/OpenLane/pdks/sky130A/libs.tech/openlane/rules.openrcx.sky130A.nom.calibre}
set ::env(RCX_RULES_MAX) {/OpenLane/pdks/sky130A/libs.tech/openlane/rules.openrcx.sky130A.max.calibre}
set ::env(RCX_RULES_MIN) {/OpenLane/pdks/sky130A/libs.tech/openlane/rules.openrcx.sky130A.min.calibre}
```

In `sky130B` PDK selection RCX extraction rules follows `spef_extractor`
```
set ::env(RCX_RULES) {/OpenLane/pdks/sky130B/libs.tech/openlane/rules.openrcx.sky130B.nom.spef_extractor}
set ::env(RCX_RULES_MAX) {/OpenLane/pdks/sky130B/libs.tech/openlane/rules.openrcx.sky130B.max.spef_extractor}
set ::env(RCX_RULES_MIN) {/OpenLane/pdks/sky130B/libs.tech/openlane/rules.openrcx.sky130B.min.spef_extractor}
```
How this `RCX_RULES` selected for `sky130A` and  `sky130B` PDK? (Configuration which control the selection)

Some of MPW designs which passed signoff time previously, after introduction of this `RCX_RULES` failed to meet signoff timing with `sky130A`.
After debugging its found that change in `RCX_RULES` for `sky130A` & `sky130B`.
So please confirm which `RCX_RULES` to use for signoff timing check.

cc/ @RTimothyEdwards  @donn

**Comment by donn** [comment, marked as answer]
@vijayank88 We picked the one that is most accurate for each variant of sky130. For sky130A, we've been able to source a calibration made with Calibre, which we believe to be the most accurate one available. For sky130B, we have no such calibration, so we fall back on the one done with `spef_extractor`.

If a design fails timing with the new calibration, it has to be updated. Reverting to the old parasitics is not acceptable.

**Reply by maliberty** [reply]
What is needed to get a similar calibration for B?

**Reply by donn** [reply]
That is a very good question. @RTimothyEdwards?

**Reply by tspyrou** [reply]
@vijayank88 how different are the numbers between the two?
